### ISSUE TITLE
/nbextensions/* get 404 response for the request url not starting with my application name

### ISSUE THREAD

**Issue by cndaimin** [jupyter/notebook#3418]
![image](https://user-images.githubusercontent.com/4527219/37269964-c63b9470-2607-11e8-99ec-bcc10f1f1f79.png)

Here app_id is for multi-user supporting, backend Nginx will omit this and dispatch the request to the specified container.
It's quite strange because other requests(like static/services/contents.js) are ok. Did I missed something?

**Comment by cndaimin** [comment]
It's ok to access them directly.

![image](https://user-images.githubusercontent.com/4527219/37270233-061b75f0-2609-11e8-9a5d-862ca0453d14.png)

**Comment by takluyver** [comment]
Looks like either `base_url` is not set, or something is not respecting it. Can you look in the HTML of the page and find `data-base-url=`?

**Comment by cndaimin** [comment]
@takluyver Thank you for your attention. `data-base-url` seems good here:

![image](https://user-images.githubusercontent.com/4527219/37285217-51266e4e-2639-11e8-8220-59d29f6322b0.png)

And path of `nbextensions`:
![image](https://user-images.githubusercontent.com/4527219/37285235-65944ebe-2639-11e8-9a4d-df9d10e35646.png)

Please let me know if you need more informations. Thanks!

**Comment by takluyver** [comment]
That's weird, those should both be constructed from the same template variable:

https://github.com/jupyter/notebook/blob/5.4.0/notebook/templates/page.html#L32

https://github.com/jupyter/notebook/blob/5.4.0/notebook/templates/notebook.html#L31

Can you find those files in your installed copy of notebook and see what the equivalent lines say?

**Comment by cndaimin** [comment]
Looks like same:
![image](https://user-images.githubusercontent.com/4527219/37318631-82f128b6-26a6-11e8-95c5-6e275aba551e.png)

One little difference: some versions of notebook.html line 31 differs from  https://github.com/jupyter/notebook/blob/5.4.0/notebook/templates/notebook.html#L31

![image](https://user-images.githubusercontent.com/4527219/37318732-1c526fce-26a7-11e8-887c-f5db56fb5583.png)

**Comment by cndaimin** [comment]
I checked the source HTML of notebook page powered by JupyterHub, the variables make sense this way:
![image](https://user-images.githubusercontent.com/4527219/37319625-df237428-26aa-11e8-984e-a0ecf2ff520c.png)
![image](https://user-images.githubusercontent.com/4527219/37319638-e8d0c2b4-26aa-11e8-8772-e28ab3a0a0dd.png)
Is it possible JupyterHub did some settings like c.NoteBookApp.x = x while I did not?

**Comment by cndaimin** [comment]
Hi takluyver: 
Here is some updates.
I hacked this line: `nbextensions : '{{ base_url }}nbextensions',` just into: `nbextensions : '../nbextensions',`(looks like the current path is `/app_xxx/static/`). Ugly but **works** for me currently.

**Comment by takluyver** [comment]
Yep, JupyterHub will be setting base_url for the individual servers.

I'm a bit confused now. Is the original bug with some other multiuser framework that's not JupyterHub?

**Comment by tristanlt** [comment]
Hi, I've the same problem with Mayavi integration inside JupyterHub. My hub was hosted behind a reverse proxy with sub-uri (https://host.fqdn/app/notebooks/)

Notebook try to get x3d.js assets from https://host.fqdn/nbextensions/....

As workaround, I want to set `local=False` but that doesn't work and, after reading notebook.py, I don't understand how this setting can work anyway...

Any news ?
Regards,
Tristan
